docs/book/src/simple-external-plugin-tutorial/api.go

```diff
@@ -0,0 +1,46 @@
+/*
```

Comment:
Could we:
- Add the plugin inside of a directory called tesdata? Why? In Golang it is a good practice to know that is a mock/sample. So that would be `docs/book/src/simple-external-plugin-tutorial/testdata/` dir instead
- Could we have a better layout for this sample , i.e.:

docs/book/src/simple-external-plugin-tutorial/testdata/
|--- pkg
         | ---> we add here api.go/ init.go, helpers.go and etc
- then we have under main.go, go.mod, go.sum

Also, have we not templates in this plugin? 
If so, could we have them instead of templates dir much more similar to how we structure our plugin? Look the pkg/plugins/golang/v3

Reply:
@camilamacedo86 I updated the structure of the sample external plugin to be more similar to the existing Kubebuilder plugins in commit https://github.com/kubernetes-sigs/kubebuilder/pull/2810/commits/e9e5c6b7f92df0b8a5956f596bb4ea2cb38da39b

Let me know what you think!